include/rapidjson/encodings.h

```diff
@@ -662,6 +694,10 @@ struct Transcoder {
         unsigned codepoint;
         if (!SourceEncoding::Decode(is, &codepoint))
             return false;
+#ifdef VALIDATE_CODEPOINT
```

Comment:
This is not ideal. Since this is a feature, I think it should be better switched on/off via non-type template parameter flag.

Reply:
Good idea, I will handle this update.